### PR TITLE
Remove AIO for sequential reads

### LINKED ISSUE #25997: `min_bytes_to_use_direct_io` works incorrectly for `max_threads` > 1
**Describe the bug**

Dataset from:
https://clickhouse.tech/docs/en/getting-started/example-datasets/metrica/
> Obtaining Tables from Prepared Partitions

**Does it reproduce on recent release?**

Yes, it is reproduced on master.

**How to reproduce**

```
SET min_bytes_to_use_direct_io = 1, max_threads = 2;

SELECT sum(cityHash64(*)) FROM test.hits

SELECT sum(cityHash64(*))
FROM test.hits

Query id: 115cee6f-7f66-42e4-a964-cf7b0db39686

↑ Progress: 8.71 million rows, 8.29 GB (2.28 million rows/s., 2.17 GB/s.) ███████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████▋    97%
0 rows in set. Elapsed: 86.030 sec. Processed 8.71 million rows, 8.29 GB (101.24 thousand rows/s., 96.37 MB/s.)
```

### PR DESCRIPTION
**Pull request by alexey-milovidov**
Changelog category (leave one):
- Improvement

Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
Remove complicated usage of Linux AIO with one block readahead and replace it with plain simple synchronous IO with O_DIRECT. In previous versions, the setting `min_bytes_to_use_direct_io` may not work correctly if `max_threads` is greater than one. Reading with direct IO (that is disabled by default for queries and enabled by default for large merges) will work in less efficient way. This closes #25997.

### CODE REVIEW

File: src/IO/createReadBufferFromFileBase.cpp [hunk 1]
```diff
@@ -20,36 +19,15 @@ namespace DB
 {
 
 std::unique_ptr<ReadBufferFromFileBase> createReadBufferFromFileBase(
-    const std::string & filename_,
-    size_t estimated_size, size_t aio_threshold, size_t mmap_threshold, MMappedFileCache * mmap_cache,
-    size_t buffer_size_, int flags_, char * existing_memory_, size_t alignment)
+    const std::string & filename,
+    size_t estimated_size, size_t direct_io_threshold, size_t mmap_threshold, MMappedFileCache * mmap_cache,
```
Comment (alesapin) [inline src/IO/createReadBufferFromFileBase.cpp:23]: Still, a confusing name since we don't use O_DIRECT?

**Review by alesapin — APPROVED** [review]
LGTM

**Comment by alexey-milovidov** [comment]
@alesapin The method with posix_fadvise was incorrect - it first asked to remove file data from page cache, then reads it with cache, potentially thrashing it. Not what we need.

I returned back O_DIRECT but with simple implementation inside ReadBufferFromFileDescriptor.

**Comment by alexey-milovidov** [comment]
It's interesting that reading becomes faster.

`clickhouse-benchmark --min_bytes_to_use_direct_io 1 <<< "SELECT sum(cityHash64(*)) FROM test.hits"`

20.8: MiB/s: 3379.232
This PR: MiB/s: 3861.583

**Comment by alexey-milovidov** [comment]
FYI we don't have automated performance tests for direct IO, because they are performed on tmpfs.

**Pull request merged**